packages/in-app-notification-manager/src/services/inAppNotificationService.ts

```diff
@@ -1,27 +1,51 @@
 import { drizzle } from "drizzle-orm/node-postgres";
-import { AppContext, UIAuthData, WithLogger } from "pagopa-interop-commons";
-import { and, eq, desc } from "drizzle-orm";
-import { notification, Notification } from "../db/schema.js";
+import {
+  AppContext,
+  createListResult,
+  UIAuthData,
+  WithLogger,
+} from "pagopa-interop-commons";
+import { and, eq, ilike, desc, getTableColumns } from "drizzle-orm";
+import {
+  fromNotificationSQL,
+  ListResult,
+  Notification,
+} from "pagopa-interop-models";
+import { withTotalCount } from "pagopa-interop-commons";
+import { notification } from "../db/schema.js";
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 export function inAppNotificationServiceBuilder(
   db: ReturnType<typeof drizzle>
 ) {
   return {
-    getNotifications: ({
-      logger,
-      authData: { userId, organizationId },
-    }: WithLogger<AppContext<UIAuthData>>): Promise<Notification[]> => {
+    getNotifications: async (
+      q: string | undefined,
+      limit: number,
+      offset: number,
+      {
+        logger,
+        authData: { userId, organizationId },
+      }: WithLogger<AppContext<UIAuthData>>
+    ): Promise<ListResult<Notification>> => {
       logger.info("Getting notifications");
-      return db
-        .select()
+      const notifications = await db
+        .select(withTotalCount(getTableColumns(notification)))
         .from(notification)
         .where(
           and(
             eq(notification.userId, userId),
-            eq(notification.tenantId, organizationId)
+            eq(notification.tenantId, organizationId),
+            q ? ilike(notification.body, `%${q}%`) : undefined
```

Comment:
my previous comment got lost because of force pushing I think, so re-adding it: we might need to escape `q` using `escapeRegExp` like in other some readmodel queries